tests/compile-fail/check_arg_count.rs

```diff
@@ -0,0 +1,8 @@
+#![feature(core_intrinsics)]
+
+use std::intrinsics;
+
+fn main() {
+    unsafe { intrinsics::forget(); } //~ ERROR this function takes 1 argument but 0 arguments were supplied
+    unsafe { intrinsics::forget(1, 2); } //~ ERROR this function takes 1 argument but 2 arguments were supplied
```

Comment:
This won't work, we stop execution after the first failure.  You have to write two tests, I am afraid.

Reply:
It seems like CI passes; I have no idea how that is possible... I'll need to investigate that. But please still make it two tests.

Reply:
Wow, something really strange is happening... I am seeing both errors with `./miri run`, whereas usually we abort on the first sight of UB...

Reply:
Oh lol. You are testing the wrong errors. :rofl: 

These errors are from the type checker. You will need to define your own wrong signature for the intrinsic and call it like that for Miri to even start doing something.

Probably it is better to do that with a foreign function than an intrinsic. See `shim_arg_size` for an example test that does something like that.